### PR TITLE
Cleanup tokenv2 collection

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Adds max sizes to URI, Description, and Name for both token and collection.  Additionally, standardizes naming of errors and adds documentation

### Test Plan
```
Test result: OK. Total tests: 59; passed: 59; failed: 0
+-------------------------+
| Move Coverage Summary   |
+-------------------------+
Module 0000000000000000000000000000000000000000000000000000000000000004::royalty
>>> % Module coverage: 83.52
Module 0000000000000000000000000000000000000000000000000000000000000004::collection
>>> % Module coverage: 83.72
Module 0000000000000000000000000000000000000000000000000000000000000004::token
>>> % Module coverage: 81.29
Module 0000000000000000000000000000000000000000000000000000000000000004::property_map
>>> % Module coverage: 86.50
Module 0000000000000000000000000000000000000000000000000000000000000004::aptos_token
>>> % Module coverage: 77.05
+-------------------------+
| % Move Coverage: 82.36  |
+-------------------------+

```

### CODE REVIEW

File: aptos-move/framework/aptos-token-objects/sources/collection.move [hunk 1]
```diff
@@ -9,34 +9,39 @@
...
+    const ETOO_MANY_TOKENS: u64 = 2;
+    /// The collection name is over the maximum length
+    const ECOLLECTION_NAME_TOO_LONG: u64 = 3;
+    /// The URI is over the maximum length
+    const EURI_TOO_LONG: u64 = 4;
+    /// The description is over the maximum length
+    const EDESCRIPTION_TOO_LONG: u64 = 5;
+
+    const MAX_COLLECTION_NAME_LENGTH: u64 = 128;
+    const MAX_URI_LENGTH: u64 = 512;
```
Comment (davidiw) [inline aptos-move/framework/aptos-token-objects/sources/collection.move:43]: 2kb, please

File: aptos-move/framework/aptos-token-objects/sources/token.move [hunk 2]
```diff
@@ -1,30 +1,36 @@
 /// This defines an object-based Token. The key differentiating features from the Aptos standard
 /// token are:
-/// * Decouple token ownership from token data.
+/// * Decoupled token ownership from token data.
```
Comment (davidiw) [inline aptos-move/framework/aptos-token-objects/sources/token.move:3]: now my brain is hurting on the grammar

**Review by davidiw — APPROVED** [review]
you really enjoy reviewing this, eh?
nice work

File: aptos-move/framework/aptos-token-objects/sources/collection.move [hunk 3]
```diff
@@ -9,34 +9,39 @@
...
 
     friend aptos_token_objects::token;
 
-    /// The collections supply is at its maximum amount
-    const EEXCEEDS_MAX_SUPPLY: u64 = 1;
     /// The collection does not exist
-    const ECOLLECTION_DOES_NOT_EXIST: u64 = 2;
+    const ECOLLECTION_DOES_NOT_EXIST: u64 = 1;
+    /// The collection has reached its supply and no more tokens can be minted
+    const ETOO_MANY_TOKENS: u64 = 2;
```
Comment (lightmark) [inline aptos-move/framework/aptos-token-objects/sources/collection.move:34]: supply_exceeded?

**Review by lightmark — APPROVED** [review]

**Pull request merged**